### PR TITLE
Enable Alternative Checksum format of algorithm_hash

### LINKED ISSUE #521: Service Binding - "dependency-mapping" will not work on Kubernetes - colon not allowed in Secret key name
Similar to https://github.com/paketo-buildpacks/packit/issues/453, sha256:b4cb31162ff6d7926dd09e21551fa745fa3ae1758c25148b48dadcf78ab0c24c as a key name cannot work on Kubernetes as ':' is not allowed as a key in a Kubernetes secret
```
$SERVICE_BINDING_ROOT
└── my-dependency-binding
    ├── type -> "dependency-mapping"
    ├── sha256:b4cb31162ff6d7926dd09e21551fa745fa3ae1758c25148b48dadcf78ab0c24c -> https://example.com/dep-1.tgz
    └── sha256:efa6d87993ff21615e2d8fc0c98e07ff357fc9f3b9bd93c2cf58ba7f2b6fd2e0 -> https://example.com/dep-2.tgz
```

link to [dependency-mapping](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0010-dependency-mappings.md#service-binding-specification-for-kubernetes-example) documentation

Motivation here is to build dotnet applications offline, which uses sha512 in their manifests.

### PR DESCRIPTION
**Pull request by ChuckQuinnIV**
## Summary

Enables an alternative format of `algorithm_hash` for specifying dependency-mappings, in addition to formats proposed in [RFC 10](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0010-dependency-mappings.md).

Resolves https://github.com/paketo-buildpacks/packit/issues/521 & https://github.com/paketo-buildpacks/packit/issues/453

## Use Cases

In Kubernetes, `:` cannot be used in a key name, meaning service-binding secrets like below break:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: dependency-mapping
type: service.binding/dependency-mapping
stringData:
  type: dependency-mapping
  ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad: https://my-internal-url-foo.example.com/dependency1.zip # works fine
  sha512:ddaf35a193617abacc417349ae20413112e6fa4e89a97ea20a9eeee64b55d39a2192992a274fc1a836ba3c23a3feebbd454d4423643ce80e2a9ac94fa54ca49f: https://my-internal-url-foo.example.com/dependency2.zip # breaks
```

Windows has a similar problem, as `:` is not allowed in file names.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Review by robdimsdale — DISMISSED** [review]
This looks fine from a code perspective - the unit tests cover the code change and I don't think we need an integration test.

I haven't thought through the ramifications of supporting this alternative hash structure but it seems reasonable. I'm not sure how else to support k8s bindings.

**Review by sophiewigmore — DISMISSED** [review]

**Comment by sophiewigmore** [comment]
I'm good with this too - we'll just need an addendum to the dependency mapping rfc to account for this case - I'll add

**Comment by sophiewigmore** [comment]
https://github.com/paketo-buildpacks/rfcs/pull/299

**Comment by sophiewigmore** [comment]
@ChuckQuinnIV thanks for the update - just want to get an 👍 across the board on the rfc and make sure no more changes are coming down the pipeline and then we can get this merged in

File: postal/internal/dependency_mappings.go [hunk 1]
```diff
@@ -50,6 +50,13 @@ func (d DependencyMappingResolver) FindDependencyMapping(checksum, platformDir s
 				return "", err
 			}
 			return strings.TrimSpace(content), nil
+			// binding provided in the form `algorithm-hash`
```
Comment (sophiewigmore) [inline postal/internal/dependency_mappings.go]: nit
```suggestion
			// binding provided in the form `algorithm_hash`
```

**Comment by sophiewigmore** [comment]
one TINY nitpick then I'm happy to merge this in. Thank you!

**Review by sophiewigmore — APPROVED** [review]

**Pull request merged**